staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -159,6 +160,9 @@ func (w *watcher) run(ctx context.Context, key string, rev int64, recursive bool
 }
 
 func (w *watcher) updatePartitionConfig(dp corev1.DataPartitionConfig) {
+	// try to avoid concurrent map writes error
```

Comment:
My concerns is that do we need to add  concurrent read protection as well while the partitionConfig is being updated?

Reply:
From the code, reads only happens after update is done. However, we still have concurrent write error. I will keep digging.